scripts/gz10/gz10.py

```diff
@@ -0,0 +1,145 @@
+        datasets.BuilderConfig(name="gz10_with_healpix_with_images",
+                               version=VERSION,
+                               data_files=DataFilesPatternsDict.from_patterns({'train': ['datafiles/healpix=*/*.h5']}),
+                               description="GZ-10 Catalog loaded with Healpix indices (using NSIDE=16). Images included.")
+    ]
+
+    DEFAULT_CONFIG_NAME = "gz10_with_healpix"  # It's not mandatory to have a default configuration. Just use one if it make sense.
+
+    _image_size = 256
+    _n_samples = 17736
+
+    def _info(self):
+        """Defines the dataset info."""
+        features = datasets.Features(
+            {
+                "gz10_label": datasets.Value("int32"),
+                "ra": datasets.Value("float32"),
+                "dec": datasets.Value("float32"),
+                "object_id": datasets.Value("string"),
+            }
+        )
+
+        if self.config.name == "gz10_images" or self.config.name == "gz10_with_healpix_with_images":
+            features["images"] = datasets.Array3D(shape=(self._image_size, self._image_size, 3), dtype="uint8")
+            features["pixel_scale"] = datasets.Value("float32")
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """We handle string, list and dicts in datafiles"""
+        if not self.config.data_files:
+            raise ValueError(f"At least one data file must be specified, but got data_files={self.config.data_files}")
+        splits = []
+        for split_name, files in self.config.data_files.items():
+            if isinstance(files, str):
+                files = [files]
+            splits.append(datasets.SplitGenerator(name=split_name, gen_kwargs={"files": files})) 
+        return splits
+
+    def _generate_examples(self, files, object_ids=None):
+        """Yeilds examples from the dataset"""
+        for j, file_path in enumerate(files):
+            with h5py.File(file_path, "r") as data:     
+                if object_ids is not None:
+                    keys = object_ids[j]
+                else:
+                    keys = data["object_id"]
+
+                # Preparing an index for fast searching through the catalog
+                sort_index = np.argsort(data["object_id"])
+                sorted_ids = data["object_id"][:][sort_index]
+
+                for k in keys:
+                    # Extract the indices of requested ids in the catalog 
+                    i = sort_index[np.searchsorted(sorted_ids, k)]
+
+                    example = {
+                        "gz10_label": data["ans"][i].astype(np.int32),
+                        "dec": data["dec"][i].astype(np.float32),
+                        "ra": data["ra"][i].astype(np.float32),
+                    }
```

Comment:
You required redshift in the HDF5 but don't read it out here. Not a problem necessarily, just curious.